mathics/builtin/distance/numeric.py

```diff
@@ -186,16 +186,16 @@ def eval(self, u, v, evaluation: Evaluation):
         "CosineDistance[u_, v_]"
 
         true_function = Expression(SymbolFunction, SymbolTrue)
-        u_is_vector = check_ArrayQ(u, Integer1, true_function, evaluation) is SymbolTrue
-        v_is_vector = check_ArrayQ(u, Integer1, true_function, evaluation) is SymbolTrue
+        u_is_vector = eval_ArrayQ(u, Integer1, true_function, evaluation) is SymbolTrue
+        v_is_vector = eval_ArrayQ(u, Integer1, true_function, evaluation) is SymbolTrue
 
         if not u_is_vector:
-            if isinstance(u, (Complex, Integer, Real)) and isinstance(
+            if not isinstance(u, (Complex, Integer, Real)) and isinstance(
                 v, (Complex, Integer, Real)
             ):
                 return
 
-            if v_is_vector:
+            elif v_is_vector:
```

Comment:
Usually linters complain about this structure, in which an `if` block ends with a `return` instruction, and then it is followed by an `elif` clause. The reason is that it is clear that the code in the`elif` block is only executed if the condition in the `if` clause is false. 
Why you want to change this?

Reply:
It looks like this had been the case in master as well. Addressed now though.